src/textual/widgets/_data_table.py

```diff
@@ -322,82 +427,233 @@ def __init__(
         classes: str | None = None,
     ) -> None:
         super().__init__(name=name, id=id, classes=classes)
+        self.data: dict[RowKey, dict[ColumnKey, CellType]] = {}
+        """Contains the cells of the table, indexed by row key and column key.
+        The final positioning of a cell on screen cannot be determined solely by this
+        structure. Instead, we must check _row_locations and _column_locations to find
+        where each cell currently resides in space."""
```

Comment:
May want reformatting to confirm with [our docstring guidelines](https://github.com/Textualize/textual/wiki/Documentation-standards#class).